### ISSUE TITLE
Problem of simultaneous usage of UDP associate and deny deny * * 127.0.0.1/8

### ISSUE THREAD

**Issue by GregoryKomissarov** [3proxy/3proxy#1022]
Hi! First of all, thank you for a good OSS product. I want to make work  UDP associate and not let proxy clients connect to the loopback interface.
If I'm putting deny before allow -  TCP over socks5 works well but not the UDP:
```
nscache 65536
nserver 8.8.8.8
nserver 8.8.4.4

config /conf/3proxy.cfg
monitor /conf/3proxy.cfg

log /logs/3proxy-%y%m%d.log D
rotate 60
counter /count/3proxy.3cf

users $/conf/passwd

include /conf/counters
include /conf/bandlimiters

auth strong
deny * * 127.0.0.1/8
allow * * * * CONNECT,UDPASSOC,BIND,HTTP,HTTPS,FTP
socks
flush
```

If I'm commenting deny section or moving after allow - UDP associate works also well.

How to make UDP associate work and at the same time not allow clients to connect to 127.0.0.1/8?

**Comment by iMiMx** [comment]
Good catch!  Been fiddling/battling with UDP myself ... as soon as I comment out (any) 'deny' for SOCKS, everything works.

Presumably you've not found a work around?  I haven't yet built from the latest master, that was going to be my next step, when I saw your post.

**Comment by z3APA3A** [comment]
Support for UDPASSOC acls added to 1.0.0 / 0.9.9.0
